src/WebJobs.Script/Binding/StreamValueBinder.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Bindings
+{
+    /// <summary>
+    /// Flexible binder that can handle automatic mappings from <see cref="Stream"/> to various other types.
+    /// </summary>
+    internal abstract class StreamValueBinder : ValueBinder
+    {
+        // collection of Types that this binder supports binding
+        // input parameters to
+        private static readonly Type[] ReadAccessTypes = new Type[]
+        {
+            typeof(Stream),
+            typeof(TextReader),
+            typeof(StreamReader),
+            typeof(string),
+            typeof(byte[])
+        };
+
+        // collection of Types that this binder supports binding
+        // output parameters to
+        private static readonly Type[] WriteAccessTypes = new Type[]
+        {
+            typeof(Stream),
+            typeof(TextWriter),
+            typeof(StreamWriter),
+            typeof(string),
+            typeof(byte[])
+        };
+
+        private readonly ParameterInfo _parameter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamValueBinder"/> class.
+        /// </summary>
+        /// <param name="parameter">The parameter being bound to.</param>
+        /// <param name="bindStepOrder">The step order</param>
+        protected StreamValueBinder(ParameterInfo parameter, BindStepOrder bindStepOrder = BindStepOrder.Default)
+            : base(parameter.ParameterType, bindStepOrder)
+        {
+            _parameter = parameter;
+        }
+
+        /// <summary>
+        /// Gets the set of Types this binder supports based on the specified <see cref="FileAccess"/>.
+        /// From the base stream, this binder will handle conversions to the other types.
+        /// </summary>
+        public static IEnumerable<Type> GetSupportedTypes(FileAccess access)
```

Comment:
It looks like this StreamValueBinder type copied from extensions exists in this repo only for access to this single static method? I don't see any classes deriving from StreamValueBinder/ValueBinder in this repo. Perhaps we can not add these two new types, and instead add just the two static helpers?